### PR TITLE
[GTests Refactoring] Removing MIOPEN_TEST_ALL and MIOPEN_TEST_FLOAT_ARG env variables. Part 5

### PR DESCRIPTION
**Pull request by Vsevolod1983**
Part 5.

Base branch for this PR is GTests_Refactoring_Integration_Branch

### CODE REVIEW

File: test/gtest/miopendriver_gemm.cpp [hunk 1]
```diff
@@ -68,13 +68,13 @@ void RunMIOpenDriver(const std::string& floatArg, const std::vector<TestCase>& t
...
 
-TEST_P(GPU_MIOpenDriverGemmTest_FP32, MIOpenDriverGemm) { RunMIOpenDriver("--float", GetParam()); };
+TEST_P(GPU_MIOpenDriverGemmTest_FP32, MIOpenDriverGemm) { RunMIOpenDriver(GetParam()); };
 
 INSTANTIATE_TEST_SUITE_P(Full,
                          GPU_MIOpenDriverGemmTest_FP32,
                          testing::Values(GetTestCases(miopendriver::basearg::gemm::Float)));
 
-TEST_P(GPU_MIOpenDriverGemmTest_FP16, MIOpenDriverGemm) { RunMIOpenDriver("--half", GetParam()); };
+TEST_P(GPU_MIOpenDriverGemmTest_FP16, MIOpenDriverGemm) { RunMIOpenDriver(GetParam()); };
```
Comment (BrianHarrisonAMD) [inline test/gtest/miopendriver_gemm.cpp:77]: I think these should be smoke tests.

ShouldRunMIOpenDriverTest with a false argument meant ignore MIOPEN_TEST_ALL, and I think that matches smoke tests now.

File: test/gtest/gtest_env_check_common.hpp [hunk 2]
```diff
@@ -25,18 +25,11 @@
...
-// conditions match expected conditions to run this test suite.
-template <typename disabled_mask, typename enabled_mask, typename check_functor>
-bool ShouldRunTestCase(check_functor&& checkConditions)
+template <typename disabled_mask, typename enabled_mask>
+bool ShouldRunTestCase()
 {
-    return IsTestSupportedForDevMask<disabled_mask, enabled_mask>() &&
-           (!MIOPEN_TEST_ALL || checkConditions());
+    return IsTestSupportedForDevMask<disabled_mask, enabled_mask>();
 }
```
Comment (BrianHarrisonAMD) [inline test/gtest/gtest_env_check_common.hpp]: I think now that we finished the clean-up this file can be removed, and all the ShouldRunTestCase() checks can be swapped to IsTestSupportedForDevMask<disabled_mask, enabled_mask>() directly.

I don't think we need to do it right now, but could simplify a bit.

File: test/gtest/miopendriver_regression_issue_1576.cpp [hunk 3]
```diff
@@ -57,7 +56,7 @@ void RunMIOpenDriver()
 {
     using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
     using d_mask = disabled<Gpu::Default>;
-    if(!ShouldRunMIOpenDriverTest<d_mask, e_mask>("--half", true))
+    if(!ShouldRunMIOpenDriverTest<d_mask, e_mask>())
```
Comment (BrianHarrisonAMD) [inline test/gtest/miopendriver_regression_issue_1576.cpp:59]: I think this one should be swapped from Smoke to Full.

Similar to above, I think ShouldRunMIOpenDriverTest with a true argument meant to check MIOPEN_TEST_ALL, and I think that matches full tests now.

Comment (Vsevolod1983) [reply test/gtest/miopendriver_regression_issue_1576.cpp:59]: Done

File: test/gtest/gtest_env_check_common.hpp (hunk 2 above)
Comment (Vsevolod1983) [reply test/gtest/gtest_env_check_common.hpp]: Done

File: test/gtest/miopendriver_gemm.cpp (hunk 1 above)
Comment (Vsevolod1983) [reply test/gtest/miopendriver_gemm.cpp:77]: Done

**Review by BrianHarrisonAMD — APPROVED** [review]
LGTM!

**Review by CAHEK7 — APPROVED** [review]

**Pull request merged**